### ISSUE TITLE
Timex.diff granularity for months is not correct in the case of February

### ISSUE THREAD

**Issue by nthock** [bitwalker/timex#497]
### Steps to reproduce

```elixir
iex> Timex.diff(~N[2019-02-28 16:16:18], ~N[2019-01-31 16:16:18], :months)
0
```
### Description of issue

`Timex.diff` doesn't seems to work in the month of February.

- What are the expected results?

Should return `1` instead of `0`.

**Comment by bitwalker** [comment]
Months are defined as 30 days in Timex, otherwise you'd get bizarre behavior like the difference between 02/01 and 01/31 being one month. Do you have an example of a well-known library that does it differently that we can compare with?

**Comment by nthock** [comment]
@bitwalker Just to clarify, wouldn't this an inconsistency between between `Timex.diff` and `Timex.shift`.

```
iex> Timex.shift(~N[2019-01-31 16:16:18], months: 1)
~N[2019-02-28 16:16:18]

iex> Timex.shift(~N[2019-01-31 16:16:18], days: 30)
~N[2019-03-02 16:16:18]
```

Seems like `Timex.shift` is able to take account that February is less than 30 days, and hence return the end of February instead of 2nd March.

**Comment by dideler** [comment]
@janpieper we noticed the same issue when our tests were failing today.

We're on 3.4.2 and the issue has been fixed in 3.5.0.

I couldn't find any release notes, but you can see the diff at https://github.com/bitwalker/timex/compare/3.4.2...3.5.0.

**Comment by bitwalker** [comment]
I believe most of the issues around shifting were fixed as of 3.6.1, unfortunately I've not required PRs/changes to update the CHANGELOG historically, but I do so now - hopefully that will help keep it up to date moving forward
